Define SSIZE_MAX conditionally?

I get the following warning if libmaxminddb is built with the C compiler from Strawberry Perl.

```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I../include -I../include -O2 -g -g -fms-extensions -MT maxminddb.lo -MD -MP -MF .deps/maxminddb.Tpo -c maxminddb.c -o maxminddb.o
maxminddb.c:26: warning: "SSIZE_MAX" redefined
 #define SSIZE_MAX INTPTR_MAX
 
In file included from c:/strawberry/c/lib/gcc/x86_64-w64-mingw32/8.3.0/include-fixed/limits.h:194,
                 from c:/strawberry/c/lib/gcc/x86_64-w64-mingw32/8.3.0/include-fixed/syslimits.h:7,
                 from c:/strawberry/c/lib/gcc/x86_64-w64-mingw32/8.3.0/include-fixed/limits.h:34,
                 from c:/strawberry/c/x86_64-w64-mingw32/include/stdlib.h:10,
                 [...]
                 from c:/strawberry/c/x86_64-w64-mingw32/include/windows.h:69,
                 from c:/strawberry/c/x86_64-w64-mingw32/include/winsock2.h:23,
                 from ../include/maxminddb.h:16,
                 from data-pool.h:4,
                 from maxminddb.c:8:
c:/strawberry/c/x86_64-w64-mingw32/include/limits.h:84: note: this is the location of the previous definition
 #define SSIZE_MAX _I64_MAX
```

--- a/Changes.md
+++ b/Changes.md
@@ -1,3 +1,9 @@
+## 1.9.1
+
+* `SSIZE_MAX` is now defined conditionally on Windows. The 1.9.0
+  release would cause a redefinition warning when compiled with MinGW.
+  Reported by Andreas Vögele. GitHub #338.
+
 ## 1.9.0 - 2024-01-09
 
 * On very large databases, the calculation to determine the search tree

--- a/src/maxminddb.c
+++ b/src/maxminddb.c
@@ -23,7 +23,9 @@
 #endif
 #include <windows.h>
 #include <ws2ipdef.h>
+#ifndef SSIZE_MAX
 #define SSIZE_MAX INTPTR_MAX
+#endif
 typedef ADDRESS_FAMILY sa_family_t;
 #else
 #include <arpa/inet.h>

